src/blockdata/block.rs

```diff
@@ -205,11 +265,14 @@ mod tests {
         assert_eq!(real_decode.header.version, 0x20000000);  // VERSIONBITS but no bits set
         assert_eq!(serialize(&real_decode.header.prev_blockhash), prevhash);
         assert_eq!(serialize(&real_decode.header.merkle_root), merkle);
+        assert_eq!(real_decode.header.merkle_root, real_decode.merkle_root());
         assert_eq!(real_decode.header.time, 1472004949);
         assert_eq!(real_decode.header.bits, 0x1a06d450);
         assert_eq!(real_decode.header.nonce, 1879759182);
         // [test] TODO: check the transaction data
 
+        assert!(real_decode.check_witness_commitment());
```

Comment:
Would be nice to check the inverse case, too.

Reply:
there are tons of incorrect cases but only one correct. covering a single incorrect case would add no value.